src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -60,16 +61,22 @@ public async Task<ISonarWebServer> CreateSonarWebServer(ProcessedArgs args, IDow
             {
                 return null;
             }
+            ISonarWebServer server;
             if (args.ServerInfo.IsSonarCloud)
             {
                 if (string.IsNullOrWhiteSpace(args.Organization))
                 {
                     logger.LogError(Resources.ERR_MissingOrganization);
+                    logger.LogWarning(Resources.WARN_DefaultHostUrlChanged);
                     return null;
                 }
-                return new SonarCloudWebServer(webDownloader, apiDownloader, serverVersion, logger, args.Organization, args.HttpTimeout);
+                server = new SonarCloudWebServer(webDownloader, apiDownloader, serverVersion, logger, args.Organization, args.HttpTimeout);
             }
-            return new SonarQubeWebServer(webDownloader, apiDownloader, serverVersion, logger, args.Organization);
+            else
+            {
+                server = new SonarQubeWebServer(webDownloader, apiDownloader, serverVersion, logger, args.Organization);
+            }
+            return await CanAuthenticate(webDownloader) ? server : null;
```

Comment:
We can also do this check after the downloader is created, before creating the server. 

This will allow to exit early and also simplify the fow a bit.

Reply:
Smart!
I didn't think of it because before we were testing on the server 🤦